chain-signatures/node/src/rpc.rs

```diff
@@ -4,27 +4,36 @@ use crate::protocol::signature::SignRequest;
 use crate::protocol::{Chain, ProtocolState};
 use crate::util::AffinePointExt as _;
 
+use alloy::primitives::Address;
+use alloy::providers::fillers::{FillProvider, JoinFill, WalletFiller};
+use alloy::providers::RootProvider;
 use cait_sith::protocol::Participant;
 use cait_sith::FullSignature;
 use k256::Secp256k1;
 use mpc_keys::hpke;
 use mpc_primitives::SignId;
 use mpc_primitives::Signature;
 
+use alloy::{
+    contract::{ContractInstance, Interface},
+    dyn_abi::DynSolValue,
+    network::EthereumWallet,
+    primitives::U256,
+    providers::ProviderBuilder,
+    transports::http::{Client as ReqwestClient, Http},
+};
```

Comment:
nit: I don't see many other cases of more-than-one levels of import group nesting. Not sure if anybody cares, I think we are already quite inconsistent with imports, which is fine by me. But you might want to check if you are using rust analyzer settings in your editor that generates this. I think if you set `imports.granularity.group` to `module`, instead of crate, it will be more in line with existing code.

https://rust-analyzer.github.io/book/features.html?highlight=import#import-granularity